distributed/tests/test_client_loop.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+import threading
+from time import sleep
+from distributed import LocalCluster, Client
+from distributed.utils import LoopRunner
+
+
+def run_client(loop):
+    with LocalCluster(loop=loop, asynchronous=False) as cluster:
+        client = Client(cluster, loop=loop, asynchronous=False)
+        sleep(1)  # Allow IO loop process events.
+        client.close()
```

Comment:
Is there a reason why this is pulled out into a separate function?  If there isn't a specific need for this I would prefer that this be included directly as normal code.

Also, you can drop the `asynchronous=False` keywords (these are normal) but I would prefer using `with Client` context managers.

`sleep(1)` makes me nervous.  Is there a specific event that you're waiting for?  Would it be enough to run a small computation like `await client.submit(inc, 1)`.  This would both be faster and also ensure that something happened.

Can we add `processes=True` to the `LocalCluster` call?  This should shave off a couple of seconds from the time of this test (we try to keep tests fast when possible).

Reply:
> Is there a reason why this is pulled out into a separate function? If there isn't a specific need for this I would prefer that this be included directly as normal code.

I used parts from py2 tests where it's in a separate function, will move to test body.

> Also, you can drop the asynchronous=False keywords (these are normal) but I would prefer using with Client context managers.

Yep, also there could be two tests: context manager and explicit, to check related methods. I'll change to use context manager to be more compact. Later will add explicit `close()` if everything will be fine.

> sleep(1) makes me nervous. Is there a specific event that you're waiting for? Would it be enough to run a small computation like await client.submit(inc, 1). This would both be faster and also ensure that something happened.

I used it to give some time to IO loop, probably better to make the test async and use `await` as you suggested.

> Can we add processes=True to the LocalCluster call?

I'll try.

Seems my changes still break some tests. I created py3.6 conda env and ran tests there. One test failed: `test_dask_worker.py:test_nanny_worker_port_range_too_many_workers_raises`. There are more test failed in CI. Not sure why it happened.

PS. I tried to run tests separately and found that if run in a different order, previously passed tests will fail:

```
# Ok
$ py.test -m "not avoid_travis" -r s --timeout-method=thread --timeout=300 --durations=20 distributed/cli/tests/test_dask_scheduler.py distributed/deploy/tests/test_ssh.py -k "test_basic or test_defaults"

# Fail
$ py.test -m "not avoid_travis" -r s --timeout-method=thread --timeout=300 --durations=20 distributed/deploy/tests/test_ssh.py distributed/cli/tests/test_dask_scheduler.py -k "test_basic or test_defaults"
```

Tested my changes and 'master`  in py 3.6/3.8 (miniconda)